### PR TITLE
route53_zone: enable check mode

### PR DESCRIPTION
**Pull request by julienvey**
##### SUMMARY

Implement check_mode in route53_zone module

##### ISSUE TYPE

 - Feature Pull Request

##### COMPONENT NAME

route53_zone

##### ANSIBLE VERSION

```
2.6.0
```

### CODE REVIEW

**Review by resmo — APPROVED** [review]
code lgtm
shipit

**Comment by felixfontein** [comment]
Cool! This is precisely the feature I found missing on Friday ;-)

File: lib/ansible/modules/cloud/amazon/route53_zone.py [hunk 1]
```diff
@@ -195,31 +195,33 @@ def create_or_update_private(module, client, matching_zones, record):
...
+                },
+                VPC={
+                    'VPCRegion': record['vpc_region'],
+                    'VPCId': record['vpc_id'],
+                },
+                CallerReference="%s-%s" % (record['name'], time.time()),
+            )
+        except (BotoCoreError, ClientError) as e:
+            module.fail_json_aws(e, msg="Could not create hosted zone")
     hosted_zone = result['HostedZone']
```
Comment (felixfontein) [inline lib/ansible/modules/cloud/amazon/route53_zone.py]: I think `result` isn't initialized when `module.check_mode` is `True`

**Review by felixfontein — CHANGES_REQUESTED** [review]
I think there's something missing. Besides that, looks good!

File: lib/ansible/modules/cloud/amazon/route53_zone.py (hunk 1 above)
Comment (julienvey) [reply lib/ansible/modules/cloud/amazon/route53_zone.py]: Fixed. Thank you for the review

**Comment by julienvey** [comment]
I've also added a few integration tests for check mode

**Review by felixfontein — APPROVED** [review]
LGTM!

**Comment by resmo** [comment]
shipit

**Review by s-hertel — APPROVED** [review]

**Pull request merged**